app/controllers/application_controller.rb

```diff
@@ -18,6 +18,15 @@ def current_vendor
     @vendor ||= Vendor.find(session[:vendor_id]) if session[:vendor_id]
   end
   
+  def current_offeror
+    params[:controller] == "providers" ? current_provider : current_vendor
```

Comment:
so now we could DRY out across these two methods extracting

```rb
def providers?
  params[:controller] == 'providers'
end
```

but only if you feel like it :-)